stdlib/test/builtin/test_string.mojo

```diff
@@ -401,6 +401,57 @@ fn test_atol_base_0() raises:
         _ = atol("0of_", base=0)
 
 
+fn test_atof() raises:
+    assert_equal(375.0, atof(String("375.f")))
+    assert_equal(1.0, atof(String("001.")))
+    assert_equal(5.0, atof(String(" 005.")))
+    assert_equal(13.0, atof(String(" 013.f  ")))
+    assert_equal(-89, atof(String("-89")))
+    assert_equal(-0.3, atof(String(" -0.3")))
+    assert_equal(-69e3, atof(String(" -69E+3  ")))
+    assert_equal(123.2e1, atof(String(" 123.2E1  ")))
+    assert_equal(23e3, atof(String(" 23E3  ")))
+    assert_equal(989343e-13, atof(String(" 989343E-13  ")))
+    assert_equal(1.123, atof(String(" 1.123f")))
+    assert_equal(0.78, atof(String(" .78 ")))
+    assert_equal(121234.0, atof(String(" 121234.  ")))
+    assert_equal(985031234.0, atof(String(" 985031234.F  ")))
```

Comment:
Could you please also implement some special values, e.g. `nan`, `inf`, `-inf`, `-0` and add tests for them?